TF-3649 add no-then rule for specs

joecan opened the pull request:
##### *To:*
@jjinux @inancsevinc @kevinzang @cansin 
##### *What:*
Adding a rule to check for then calls in specs.
##### *JIRA:*
https://udemyjira.atlassian.net/browse/TF-3649
##### *What did you test:*
yarn test
##### *What dashboards will you be monitoring:*
N/A

joecan committed:
TF-3649 add no-then rule for specs

jjinux commented on packages/eslint-plugin-udemy/rules/no-then/README.md:
````diff
@@ -0,0 +1,21 @@
+
+```js
+await promise;
+```
````
It may make sense to add a comment that says:

As with a lot of rules in ESLint, there are valid times when you'll want to break this rule, but you can treat each such situation on a case-by-case basis by using eslint-disable-next-line, etc.

joecan replied:
I added this.

jjinux commented on packages/eslint-plugin-udemy/rules/no-then/index.js:
```diff
@@ -0,0 +1,17 @@
+'use strict';
+module.exports.rules = {
```
nit: blank line after 'use strict'.

jjinux commented on packages/eslint-plugin-udemy/rules/no-then/index.js:
```diff
@@ -0,0 +1,17 @@
+                    if (node.property && node.property.name === 'then') {
+                        context.report({
+                            node,
+                            message: 'Found `then` used in a spec. Please use async / await instead.',
```
The error message talks about specs, but the rule is not specific to specs.

jjinux replied:
Perhaps you should change the error to:

...You should usually use async / await instead.

jjinux commented on packages/eslint-config-udemy-website/index.js:
```diff
@@ -111,6 +111,7 @@ module.exports = {
             files: '**/*.spec.js',
             rules: {
                 'udemy/no-non-zero-settimeout': ['error', 'always'],
+                'udemy/no-then': ['error', 'always'],
```
Remember to follow the Conventual Commits standard for your commit message. This is a non-breaking change.

joecan committed:
TF-3649 fix: change wording per JJ review

joecan committed:
TF-3649 fix: add missing no-then link

joecan committed:
Publish

 - babel-polyfill-udemy-website@8.0.2
 - babel-preset-udemy-website@9.0.2
 - eslint-config-udemy-basics@6.0.2
 - eslint-config-udemy-website@8.0.2
 - eslint-plugin-udemy@7.0.2
